File: clients/PHP-client/src/JobsClient.php
<?php

namespace SoliDB;

class JobsClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function listQueues(): array
    {
        $res = $this->client->sendCommand('list_queues', ['database' => $this->client->getDatabase()]);
        return $res['data'] ?? [];
    }

    public function listJobs(string $queueName, ?string $status = null, ?int $limit = null, ?int $offset = null): array
    {
        $params = [
            'database' => $this->client->getDatabase(),
            'queue_name' => $queueName
        ];
        if ($status !== null) {
            $params['status'] = $status;
        }
        if ($limit !== null) {
            $params['limit'] = $limit;
        }
        if ($offset !== null) {
            $params['offset'] = $offset;
        }
        $res = $this->client->sendCommand('list_jobs', $params);
        return $res['data'] ?? [];
    }

    public function enqueue(string $queueName, string $scriptPath, ?array $params = null, ?int $priority = null, ?string $runAt = null): array
    {
        $args = [
            'database' => $this->client->getDatabase(),
            'queue_name' => $queueName,
            'script_path' => $scriptPath
        ];
        if ($params !== null) {
            $args['params'] = $params;
        }
        if ($priority !== null) {
            $args['priority'] = $priority;
        }
        if ($runAt !== null) {
            $args['run_at'] = $runAt;
        }
        $res = $this->client->sendCommand('enqueue_job', $args);
        return $res['data'] ?? [];
    }

    public function cancel(string $jobId): void
    {
        $this->client->sendCommand('cancel_job', [
            'database' => $this->client->getDatabase(),
            'job_id' => $jobId
        ]);
    }

    public function get(string $jobId): array
    {
        $res = $this->client->sendCommand('get_job', [
            'database' => $this->client->getDatabase(),
            'job_id' => $jobId
        ]);
        return $res['data'] ?? [];
    }
}

File: clients/Ruby-client/lib/solidb/jobs_client.rb
module SoliDB
  class JobsClient
    def initialize(client)
      @client = client
    end

    def list_queues
      @client.send_command("list_queues", database: @client.database) || []
    end

    def list_jobs(queue_name, status: nil, limit: nil, offset: nil)
      params = { database: @client.database, queue_name: queue_name }
      params[:status] = status if status
      params[:limit] = limit if limit
      params[:offset] = offset if offset
      @client.send_command("list_jobs", params) || []
    end

    def enqueue(queue_name, script_path:, params: nil, priority: nil, run_at: nil)
      args = {
        database: @client.database,
        queue_name: queue_name,
        script_path: script_path
      }
      args[:params] = params if params
      args[:priority] = priority if priority
      args[:run_at] = run_at if run_at
      @client.send_command("enqueue_job", args)
    end

    def cancel(job_id)
      @client.send_command("cancel_job", database: @client.database, job_id: job_id)
      nil
    end

    def get(job_id)
      @client.send_command("get_job", database: @client.database, job_id: job_id)
    end
  end
end
